### PR TITLE
Use `user_macros.GNU` to Find `opendds_idl` in KeyTest

### PR DESCRIPTION
**Pull request by iguessthislldo**
- Rewrote KeyTest `run_test.pl` to try to be consistent and robust.
- `bin/PerlDDS/Run_Test.pm`:
  - Made common code for finding executables.
  - Added strict and warnings and deal with the results of that.
- Fix Some Minor Issues with `lint.pl`:
  - Fix error messages being out of order on GHA by always flushing `stdout`.
  - Fix filename in `ace_condition` message.
  - Ignore files if git ignores them.

### CODE REVIEW

File: bin/PerlDDS/Run_Test.pm [hunk 1]
```diff
@@ -993,7 +1049,7 @@ sub _create_process {
     my $possible_would_be = ($self->{add_pending_timeout} ? "" : "would be ");
     my $prevent_or_allow = ($self->{add_pending_timeout} ? "prevent" : "allow");
     $self->_info("TestFramework::_create_process " . $possible_would_be
-      . "adding \"$flag\" to $executable's parameters. To " . $prevent_or_allow
+      . "adding \"$flag\" to $executable\'s parameters. To " . $prevent_or_allow
```
Comment (mitza-oci) [inline bin/PerlDDS/Run_Test.pm:1052]: I don't think this is needed (escaping `'` within `""`)

File: bin/PerlDDS/Run_Test.pm [hunk 2]
```diff
@@ -668,15 +727,12 @@ sub setup_discovery {
...
-    $executable = "$ENV{DDS_ROOT}/bin/DCPSInfoRepo";
-    if (!(-e $executable) && !(-e "${executable}.exe")) {
-      if (!defined($ENV{OPENDDS_INSTALL_PREFIX})) {
-        print STDERR "ERROR: Couldn't find \$DDS_ROOT/bin/DCPSInfoRepo. It " .
-          "needs to be built or \$OPENDDS_INSTALL_PREFIX needs to be defined " .
-          "if OpenDDS is installed.\n";
-        exit 1;
-      }
-      $executable = "$ENV{OPENDDS_INSTALL_PREFIX}/bin/DCPSInfoRepo";
+    $executable = PerlDDS::get_bin_exe("DCPSInfoRepo");
```
Comment (mitza-oci) [inline bin/PerlDDS/Run_Test.pm:730]: The CI builds are showing errors trying to find DCPSInfoRepo

File: tests/DCPS/KeyTest/.gitignore [hunk 3]
```diff
@@ -61,3 +61,6 @@
 /BoundTest2TypeSupportS.h
 /BoundTest2TypeSupportS.inl
 /IsBounded
+/invalid_*TypeSupport.idl
+/invalid_*TypeSupportImpl.cpp
+/invalid_*TypeSupportImpl.h
```
Comment (mitza-oci) [inline tests/DCPS/KeyTest/.gitignore:66]: I would rather have these ignores generated by the script, but maybe that's a discussion for another time / another PR.

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: bin/PerlDDS/Run_Test.pm (hunk 1 above)
Comment (iguessthislldo) [reply bin/PerlDDS/Run_Test.pm:1052]: According to this it does:

```
Old package separator used in string at .../bin/PerlDDS/Run_Test.pm line 1052.
	(Did you mean "$executable\'s" instead?)
Use of uninitialized value $executable::s in concatenation (.) or string at .../bin/PerlDDS/Run_Test.pm line 1054.
```

Comment (mitza-oci) [reply bin/PerlDDS/Run_Test.pm:1052]: Ah, I missed that part.  It might be more clear to use the `${var}` interpolation style.  Either way is fine.

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**